src/main/java/de/tum/in/www1/jenkins/notifications/SendTestResultsNotificationPostBuildTask.java

```diff
@@ -170,9 +167,17 @@ private List<Commit> findCommits(Run<?, ?> run) {
             final String[] urlString = buildData.getRemoteUrls().iterator().next().split("/");
             final String slug = urlString[urlString.length - 1].split("\\.")[0];
             final String hash = Objects.requireNonNull(buildData.getLastBuiltRevision()).getSha1().name();
+            final String branchName = buildData.getLastBuiltRevision().getBranches().stream().map(GitObject::getName).findFirst().orElse(null);
             final Commit commit = new Commit();
             commit.setRepositorySlug(slug);
             commit.setHash(hash);
+            if (branchName != null) {
+                // The branch name is in the format REPO_NAME/BRANCH_NAME -> We want to get rid of the REPO_NAME (the BRANCH_NAME might also contain /, so we can not simply use
+                // branchNameParts[1])
+                String[] branchNameParts = branchName.split("/");
+                String[] branchNamePartsWithoutRepositoryName = Arrays.copyOfRange(branchNameParts, 1, branchNameParts.length);
+                commit.setBranchName(String.join("/", branchNamePartsWithoutRepositoryName));
+            }
```

Comment:
I think it would be nicer to move all the branch extraction code into a separate helper method.
```suggestion
            final Commit commit = new Commit();
            commit.setRepositorySlug(slug);
            commit.setHash(hash);
            
            String branchName = extractBranchName(buildData);
            commit.setBranchName(branchName);
```

Reply:
Thanks, I addressed the change